internal/os/smb/api.go

```diff
@@ -14,8 +15,40 @@ func New() APIImplementor {
 	return APIImplementor{}
 }
 
-func (APIImplementor) NewSmbGlobalMapping(remotePath, username, password string) error {
-	klog.V(4).Infof("NewSmbGlobalMapping: remotePath:%q", remotePath)
+func (APIImplementor) IsSmbMapped(remotePath string) (bool, error) {
+	cmdLine := fmt.Sprintf(`$(Get-SmbGlobalMapping -RemotePath $Env:smbremotepath -ErrorAction Stop).Status `)
+	cmd := exec.Command("powershell", "/c", cmdLine)
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("smbremotepath=%s", remotePath))
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return false, fmt.Errorf("error checking smb mapping %s, err: %v", remotePath, err)
+	}
+
+	if len(out) == 0 || !strings.EqualFold(strings.TrimSpace(string(out)), "OK") {
```

Comment:
if there is no output, does it mean it is not mapped?

Reply:
Yes, that is what is going to be the best path in that case.